go/vt/vtgate/engine/primitive.go

```diff
@@ -42,10 +42,17 @@ const ListVarName = "__vals"
 type VCursor interface {
 	// Context returns the context of the current request.
 	Context() context.Context
+
+	// V3 functions.
```

Comment:
For the newcomer to the code it would be great to provide some more context on what "V3" means -- something like "Routing functions used by Vindex operations" would be clearer than "V3".

Perhaps you eventually plan to have the VCursor used in other contexts, but AFAIK `Execute` (and now `ExecuteAutocommit`) are only used by the vindex implementations so having this be clear in the interface would definitely help those of us who have tried to wade through what's going on here.